File: algorithms/java/src/test/java/leetcode/LC258AddDigits.java
package leetcode;

import org.junit.jupiter.api.Test;


import static org.junit.jupiter.api.Assertions.assertEquals;

/**
 * https://leetcode.com/problems/add-digits/
 * #easy
 */
public final class LC258AddDigits {
    public final class Solution {
        public int addDigits(final int num) {
            while (num >= 10) {
                int sum = 0;
                while (num > 0) {
                    int digit = num % 10;
                    sum += digit;
                    num /= 10;
                }
                num = sum;
            }
            return num;
            //return (int) (num - 9 * Math.floor((num - 1) / 9));
        }
    }

    @Test
    public void test_example() throws Exception {
        assertEquals(2, new Solution().addDigits(38));
    }
}

File: algorithms/code/leetcode/lc258_add_digits/lc258_add_digits.py
#!/usr/bin/env python3
# https://leetcode.com/problems/add-digits/

import unittest


class Solution:
    def addDigits(self, num: int) -> int:
        while num >= 10:
            summed = 0
            while num > 0:
                digit = num % 10
                summed += digit
                num = num // 10
            num = summed
        return num
        # return num - 9 * (num - 1) // 9


class TestCode(unittest.TestCase):
    def test_example(self):
        self.assertEqual(2, Solution().addDigits(38))
